### ISSUE TITLE
Introduce `decorators` argument in `module()`

### ISSUE THREAD

**Issue by gogonzo** [insightsengineering/teal#1368]
Blocked by https://github.com/insightsengineering/teal/issues/1383

Add a `decorators = list()` argument to the `module()` function and pass decorators to the `server_args` and `ui_args`. Revisit the documentation of `teal_transform_module` and `teal_data_module` and focus that it is clear:
- `teal_data_module()` is to create the teal_data during app run
- `teal_transform_module()` is to modify data, they can be used in two ways:
    - `transformers`: to modify teal_module's data input
    - `decorators`: to modify teal_modules's output objects

**Comment by donyunardi** [comment]
Hi @averissimo @llrs-roche

Should this be closed by #1357 ?

**Comment by llrs-roche** [comment]
Yes `decorators` is a parameter on srv_* and passed to the dots in ui_, resulting in module specific decorators. This results on code similar to this:

```r
app <- init(
  data = data,
  modules = modules(
    tm_module(decorators = list_decorators, ...)
  )
) 
```
